connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -466,18 +473,41 @@ private void stopTasks(Collection<ConnectorTaskId> ids) {
     }
 
     private void awaitStopTask(ConnectorTaskId taskId, long timeout) {
-        WorkerTask task = tasks.remove(taskId);
+        WorkerTask task;
+        Future<?> future;
+
+        synchronized (taskAndThreadLock) {
+            task = tasks.remove(taskId);
+            future = threads.remove(taskId);
```

Comment:
futures would match future, right ?
The ConcurrentMap stores Futures.

Reply:
Sure, will change.